test/fixtures_redis_test.go

```diff
@@ -0,0 +1,135 @@
+//+build integration
+
+package test_test
+
+import (
+	"github.com/applike/gosoline/pkg/fixtures"
+	"github.com/applike/gosoline/pkg/mdl"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/applike/gosoline/pkg/test"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/go-redis/redis"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"testing"
+	"time"
+)
+
+type RedisTestModel struct {
+	Name string `json:"name"`
+	Age  uint   `json:"age"`
+}
+
+type FixturesRedisSuite struct {
+	suite.Suite
+	client *redis.Client
+	logger mon.Logger
+}
+
+func (s *FixturesRedisSuite) SetupSuite() {
+	setup(s.T())
+	test.Boot("test_configs/config.redis.test.yml")
+	s.client = test.ProvideRedisClient("redis")
+	s.logger = mon.NewLogger()
+}
+
+func (s *FixturesRedisSuite) TearDownSuite() {
+	test.Shutdown()
+}
+
+func TestFixturesRedisSuite(t *testing.T) {
+	suite.Run(t, new(FixturesRedisSuite))
+}
+
+func (s FixturesRedisSuite) TestRedis() {
+	config := configFromFiles("test_configs/config.redis.test.yml", "test_configs/config.fixtures_redis.test.yml")
+
+	loader := fixtures.NewFixtureLoader(config, s.logger)
+
+	err := loader.Load(redisFixtures())
+	assert.NoError(s.T(), err)
+
+	setValue, err := s.client.Get("set_test").Result()
+
+	// should have created the item
+	assert.NoError(s.T(), err)
+	assert.Equal(s.T(), "bar", setValue)
+
+	rpopValue, err := s.client.LPop("rpush_test").Result()
+
+	// should have created the item
+	assert.NoError(s.T(), err)
+	assert.Equal(s.T(), "bar", rpopValue)
+
+	rpopValue, err = s.client.LPop("rpush_test").Result()
+
+	// should have created the item
+	assert.NoError(s.T(), err)
+	assert.Equal(s.T(), "baz", rpopValue)
+}
+
+func (s FixturesRedisSuite) TestRedisKvStore() {
+	config := configFromFiles("test_configs/config.redis.test.yml", "test_configs/config.fixtures_redis.test.yml")
+
+	loader := fixtures.NewFixtureLoader(config, s.logger)
+
+	err := loader.Load(redisKvstoreFixtures())
+	assert.NoError(s.T(), err)
+
+	res, err := s.client.Get("gosoline-integration-test-test-application-kvstore-testModel-kvstore_entry_1").Result()
+
+	// should have created the item
+	assert.NoError(s.T(), err)
+	assert.JSONEq(s.T(), `{"name":"foo","age":123}`, res)
+}
+
+func redisFixtures() []*fixtures.FixtureSet {
+	return []*fixtures.FixtureSet{
+		{
+			Enabled: true,
+			Writer:  fixtures.RedisFixtureWriterFactory(aws.String("default"), aws.String(fixtures.RedisOpSet)),
+			Fixtures: []interface{}{
+				[]interface{}{
```

Comment:
I think we can improve the types here.
Instead of having `[]interface{}` we could have a proper struct. Something like this:
```go
type RedisFixture struct {
    Key: string,
    Value: interface{},
    Duration: time.Duration
}
```

Reply:
👍 i put that (slightly altered) into the redis fixture writer file